### ISSUE TITLE
[suggestion] make `expect { }` accept a suspending lambda

### ISSUE THREAD

**Issue by cjbrooks12** [robfletcher/strikt#107]
The only way to execute a coroutine inside an `expect` block is to wrap the coroutine calls in `runBlocking<Unit> {`. The following block will not compile.

```
expect {
	suspendingFunction1()
	suspendingFunction2()
	suspendingFunction3()
}
```

but works like the following: 

```
expect {
    runBlocking<Unit> {
	    suspendingFunction1()
    	suspendingFunction2()
	    suspendingFunction3()
    }
}
```

But that's kind of ugly in a test. It would be nice to have a version of the `expect` callback that accepts a _suspending_ lambda, so coroutines can be executed directly within the lambda. 

The signature would look like 
```
fun expect(block: suspend ExpectationBuilder.() -> Unit) {
...
}
```

**Comment by robfletcher** [comment]
Does this have any implications for the behavior of assertion subjects within the block?

**Comment by cjbrooks12** [comment]
I'll do some testing with it, but it should work just the same. Running non-suspending functions from within a suspending function should work exactly the same as running outside one.

**Comment by robfletcher** [comment]
Fixed in a7fe10bd1b348af0a1e760b023a7bd9fd36dbbb6
